plugins/modules/s3_bucket.py

```diff
@@ -1053,6 +1143,86 @@ def handle_bucket_object_lock_retention(s3_client, module: AnsibleAWSModule, nam
     return object_lock_default_retention_changed, object_lock_default_retention_result
 
 
+def handle_bucket_inventory(s3_client, module: AnsibleAWSModule, name: str) -> tuple[bool, dict]:
+    """
+    Manage inventory configuration for an S3 bucket.
+    Parameters:
+        s3_client (boto3.client): The Boto3 S3 client object.
+        module (AnsibleAWSModule): The Ansible module object.
+        name (str): The name of the bucket to handle inventory for.
+    Returns:
+        A tuple containing a boolean indicating whether inventory settings were changed
+        and a dictionary containing the updated inventory.
+    """
+    declared_inventories = module.params.get("inventory")
+    results = []
+    bucket_changed = False
+
+    try:
+        present_inventories = _list_bucket_inventory_configurations(s3_client, name)
+    except is_boto3_error_code(["NotImplemented", "XNotImplemented"]) as e:
+        if declared_inventories is not None:
+            module.fail_json_aws(e, msg="Fetching bucket inventories is not supported")
+    except is_boto3_error_code("AccessDenied") as e:  # pylint: disable=duplicate-except
+        if declared_inventories is not None:
+            module.fail_json_aws(e, msg="Permission denied fetching bucket inventories")
+
+    except (
+        botocore.exceptions.BotoCoreError,
+        botocore.exceptions.ClientError,
+    ) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to fetch bucket inventories")
+    else:
+        if not declared_inventories and present_inventories != []:
```

Comment:
this whole section should be wrapped in an
`if declared_inventories is not None:`

We should only be making a change if someone explicitly passed a value for the parameter.